### PR TITLE
haproxy needs to request privilege escalation in order to work

### PR DESCRIPTION
**Pull request by stlaz**
This requires access to the legacy "restricted" SCC.

/assign @Miciah 
/cc @chadcrum

### CODE REVIEW

File: test/extended/router/h2spec.go [hunk 1]
```diff
@@ -84,6 +84,9 @@ var _ = g.Describe("[sig-network-edge][Conformance][Area:Networking][Feature:Rou
 			canaryImage, err := getCanaryImage(oc)
 			o.Expect(err).NotTo(o.HaveOccurred())
 
+			err = oc.AsAdmin().Run("adm", "policy", "add-scc-to-user").Args("restricted", "-n", oc.Namespace(), "-z", "default").Execute()
```
Comment (deads2k) [inline test/extended/router/h2spec.go:87]: how is this cleaned up at the end of the e2e run?

Comment (deads2k) [reply test/extended/router/h2spec.go:87]: > how is this cleaned up at the end of the e2e run?

How do you make sure the kube-apiserver observes this before creating the pod.

**Comment by deads2k** [comment]
this fix needs adjusting, but there's a chance it improves my current state which is full broke.

File: test/extended/router/h2spec.go (hunk 1 above)
Comment (stlaz) [reply test/extended/router/h2spec.go:87]: > how is this cleaned up at the end of the e2e run?
it's a rolebinding, it gets removed on NS removal

> How do you make sure the kube-apiserver observes this before creating the pod.
currently, we don't

**Pull request merged**